### PR TITLE
✨Sort Opened Solution in Specific Order

### LINKED ISSUE #1242: ✨Sort Opened Solutions
#### Describe your issue

Currently the persisted Solutions are added in no particular order. This way the remote Solutions are mixed with the filesystem Solutions.

#### How can others reproduce the issue?

- Open BPMN-Studio.
- Open a second remote solution.
- Open a filesystem solution.
- Close and reopen the studio.

#### Possible solution

Sort the opened Solutions in a way that remote solutions are always on top.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by Vyperus**
This adds a sorting mechanism to the opened Solutions, the order is:
1. Single Diagrams
2. The internal PE
3. All other remote Solutions
4. All local Solutions

**Changes:**

1. Add sorting the opened Solutions.

**Issues:**

Closes #1242

PR: #1304

## How can others test the changes?

1. Start BPMN-Studio.
2. Open some different Types of Solutions.
3. Restart the Studio, see that the Solutions stay in order.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 1]
```diff
@@ -226,7 +226,20 @@ export class SolutionExplorerList {
     const filteredEntries: Array<ISolutionEntry> = this._openedSolutions
       .filter(this._shouldDisplaySolution);
 
-    return filteredEntries;
+    const sortedEntries: Array<ISolutionEntry> = filteredEntries.sort((solutionA: ISolutionEntry, solutionB: ISolutionEntry) => {
+      if (solutionA.isSingleDiagramService) {
+        return -1;
+      }
+
+      if (solutionA.uri === window.localStorage.getItem('InternalProcessEngineRoute')) {
```
Comment (robpal94) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]: Extract this evaluation rule

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 2]
```diff
@@ -226,7 +226,21 @@ export class SolutionExplorerList {
...
+
+      const solutionAIsInternalProcessEngine: boolean = solutionA.uri === window.localStorage.getItem('InternalProcessEngineRoute');
+      if (solutionAIsInternalProcessEngine) {
+        return -1;
+      }
+
+      return solutionA.uri.startsWith('http') && !solutionB.uri.startsWith('http')
+              ? -1
+              : 1;
+    });
```
Comment (SteffenKn) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]: Add some spacing:

```suggestion
    });

```

**Comment by SteffenKn** [comment]
Imho it feels better if the deployed diagrams are above the local diagrams, because then it is easier to distinguish between deployed and local diagrams.

I would prefer this order:
```
1. The internal PE
2. All other remote Solutions
3. Single Diagrams
4. All local Solutions
```

**Comment by Vyperus** [comment]
@SteffenKn I like that the single diagrams are above the rest. Since VSCode is handling it kinda the same way.

**Comment by SteffenKn** [comment]
@Vyperus Not really. VSCode handles single files via 'Open Editors'.
But we don't support something like that atm.
@robpal94 created a dummy in [this PR](https://github.com/process-engine/bpmn-studio/pull/967), but this was never really implemented.

**Pull request merged**